client/components/App.js

```diff
@@ -13,30 +13,43 @@ import {
   getOpenTranscriptButton,
 } from "../utils/custom-buttons";
 import { GlobalStyles } from "./GlobalStyles";
-import { CopyRoomURLButton } from "./CopyRoomURLButton";
 import { useRouter } from "next/router";
 import { AIAssistant } from "./AIAssistant";
 import { Transcript } from "./Transcript";
+import { CopyContentButton } from "./CopyContentButton";
 
 export default function App() {
   const { query } = useRouter();
   const [url, setUrl] = useState("");
+  const [meetingToken, setMeetingToken] = useState("");
   const [daily, setDaily] = useState(null);
   const [isJoining, setIsJoining] = useState(false);
   const [aiView, setAiView] = useState(null);
   const wrapperRef = useRef(null);
 
-  const joinRoom = useCallback(async (url) => {
+  const joinRoom = useCallback(async (url, dailyKey, oaiKey, wantBotToken) => {
     setIsJoining(true);
-    const response = await fetch("/api/create-session", {
-      method: "POST",
-      body: JSON.stringify({
-        room_url: url,
-      }),
-    });
-    const body = await response.json();
-    if (body.url) {
-      setUrl(body.url);
+    if (url && dailyKey && oaiKey) {
+      const response = await fetch("/api/create-session", {
+        method: "POST",
+        body: JSON.stringify({
+          room_url: url,
+          daily_api_key: dailyKey,
+          want_bot_token: wantBotToken,
+          openai_api_key: oaiKey,
+        }),
+      });
+      const body = await response.json();
```

Comment:
Yes, and we also need more error handling from the server end when a request fails (since there will no longer be an error response from an HTTP request with everything being an app message). I'll do this as  a followup PR